### ISSUE TITLE
Header line too long

### ISSUE THREAD

**Issue by jjeromee** [Qarj/WebInject#41]
Hi,

I would like to know if we could increase the max header line Value ?

Currently the limit setting is 8192, I need to check webpage with a bigger value.

you can see the error log below : 
Header line too long (XXXX; limit is 8192) at /usr/local/share/perl5/Net/HTTP/Methods.pm line 284.

Best Regards.

Jérôme

**Comment by Qarj** [comment]
Hi Jérôme,

I spent ten minutes on google trying to see how to change this, but didn't find anything. Maybe you want to give it a go?

The limit is not being set by WebInject, you can see from the error message it is coming from Net::HTTP.

If you look on cpan for Net::HTTP you can see indeed that the default line length is indeed 8192.

However WebInject is using LWP and HTTP::Request::Common and in turn these are using Net::HTTP. I can't see how to configure this value, but maybe there is a way if you want to see some research and find an example?

Best Regards,

Tim

**Comment by jjeromee** [comment]
Hi Tim,

Yes you're right, the error message is coming from Net:HTTP

I've already check some example, without any success :-(

Thanks for your reply.

Jérôme

**Comment by Qarj** [comment]
Hi Jérôme,

Is it a request or response header line length limit that is the problem? If it is a response header, maybe this will work:

Change the line in webinject.pl that reads:

```
    $useragent = LWP::UserAgent->new(keep_alive=>1);
```

to

```
    $useragent = LWP::UserAgent->new(keep_alive=>1, max_line_length => 16384);
```

then let me know....

Tim

**Comment by jjeromee** [comment]
Hi Tim,

The issue stills remain even with this setting.

I've just try with an other setting, HTTP::Headers->new(content_length=>16384); , but the issue remains too.

Thanks for your reply.

Jérôme

**Comment by jjeromee** [comment]
Hi Tim,

To bypass this limit we can add those two lines just after the start_session session : 

use LWP::Protocol::http;
push(@LWP::Protocol::http::EXTRA_SOCK_OPTS, MaxLineLength => 0);

So, the issue is resolved whith those two lines.

Do you think those lines could do some other issue ?

Thanks for your reply.

Jérôme

**Comment by Qarj** [comment]
Thanks Jérôme,

I'll try out your solution and so long as it doesn't break any of our existing tests then I'll add it into the project. You did well to track down that solution!

Best Regards,

Tim

**Comment by Qarj** [comment]
released in v2.8.0
